From 16ab43db6ed6f71424d5ad78e62f85baaeae5051 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Wed, 10 Dec 2025 12:12:47 +0100
Subject: [PATCH 6/7] conf: USB-Audio: define pcm configuration block only one
 time

There may be multiple USB soundcards in the system. Overwrite
the PCM configurations when loaded multiple times.

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 src/conf/cards/USB-Audio.conf | 30 +++++++++++++++---------------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/src/conf/cards/USB-Audio.conf b/src/conf/cards/USB-Audio.conf
index 2f6d2ee0af3d..1fc540e2eb4d 100644
--- a/src/conf/cards/USB-Audio.conf
+++ b/src/conf/cards/USB-Audio.conf
@@ -99,7 +99,7 @@ USB-Audio.pcm.iec958_2_device {
 # device 0: analog output, digital input
 # device 1: digital output, analog input
 USB-Audio."AudioPhile".pcm.default "cards.USB-Audio.Audiophile USB (tm).pcm.default"
-USB-Audio."Audiophile USB (tm)".pcm.default {
+USB-Audio."Audiophile USB (tm)".pcm.!default {
 	@args [ CARD ]
 	@args.CARD { type string }
 	type asym
@@ -120,7 +120,7 @@ USB-Audio."Audiophile USB (tm)".pcm.default {
 	}
 }
 USB-Audio."AudioPhile".pcm.iec958 "cards.USB-Audio.Audiophile USB (tm).pcm.iec958"
-USB-Audio."Audiophile USB (tm)".pcm.iec958 {
+USB-Audio."Audiophile USB (tm)".pcm.!iec958 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -142,7 +142,7 @@ USB-Audio."Audiophile USB (tm)".pcm.iec958 {
 
 # For this card we can (and must to get IEC61937) set AES bits
 USB-Audio."MicroII".pcm.iec958 "cards.USB-Audio.Audio Advantage MicroII.pcm.iec958"
-USB-Audio."Audio Advantage MicroII".pcm.iec958 {
+USB-Audio."Audio Advantage MicroII".pcm.!iec958 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -180,7 +180,7 @@ USB-Audio."Audio Advantage MicroII".pcm.iec958 {
 
 <confdir:pcm/front.conf>
 
-USB-Audio.pcm.front.0 {
+USB-Audio.pcm.front.!0 {
 	@args [ CARD ]
 	@args.CARD { type string }
 	@func refer
@@ -201,7 +201,7 @@ USB-Audio.pcm.front.0 {
 	}
 }
 
-USB-Audio.pcm.default {
+USB-Audio.pcm.!default {
 	@args [ CARD ]
 	@args.CARD { type string }
 	@func refer
@@ -249,14 +249,14 @@ USB-Audio.pcm.default {
 	}
 }
 
-USB-Audio.pcm.default_playback_dmix_yes {
+USB-Audio.pcm.!default_playback_dmix_yes {
 	@args [ CARD ]
 	@args.CARD { type string }
 	@func concat
 	strings [ "dmix:" $CARD ]
 }
 
-USB-Audio.pcm.default_playback_dmix_no {
+USB-Audio.pcm.!default_playback_dmix_no {
 	@args [ CARD ]
 	@args.CARD { type string }
 	type hw
@@ -266,7 +266,7 @@ USB-Audio.pcm.default_playback_dmix_no {
 
 <confdir:pcm/surround40.conf>
 
-USB-Audio.pcm.surround40.0 {
+USB-Audio.pcm.surround40.!0 {
 	@args [ CARD ]
 	@args.CARD { type string }
 	@func refer
@@ -301,7 +301,7 @@ USB-Audio.pcm.surround40.0 {
 	}
 }
 
-USB-Audio.pcm.surround40_default {
+USB-Audio.pcm.!surround40_default {
 	@args [ CARD ]
 	@args.CARD { type string }
 	type hw
@@ -309,7 +309,7 @@ USB-Audio.pcm.surround40_default {
 	device 0
 }
 
-USB-Audio.pcm.surround40_six_channels {
+USB-Audio.pcm.!surround40_six_channels {
 	@args [ CARD ]
 	@args.CARD { type string }
 	type route
@@ -327,7 +327,7 @@ USB-Audio.pcm.surround40_six_channels {
 	}
 }
 
-USB-Audio.pcm.surround40_two_stereo_devices {
+USB-Audio.pcm.!surround40_two_stereo_devices {
 	@args [ CARD ]
 	@args.CARD { type string }
 	type route
@@ -369,7 +369,7 @@ USB-Audio.pcm.surround40_two_stereo_devices {
 <confdir:pcm/surround50.conf>
 <confdir:pcm/surround51.conf>
 
-USB-Audio.pcm.surround51.0 {
+USB-Audio.pcm.surround51.!0 {
 	@args [ CARD ]
 	@args.CARD { type string }
 	@func refer
@@ -402,7 +402,7 @@ USB-Audio.pcm.surround51.0 {
 
 <confdir:pcm/surround71.conf>
 
-USB-Audio.pcm.surround71.0 {
+USB-Audio.pcm.surround71.!0 {
 	@args [ CARD ]
 	@args.CARD { type string }
 	@func refer
@@ -437,7 +437,7 @@ USB-Audio.pcm.surround71.0 {
 
 <confdir:pcm/iec958.conf>
 
-USB-Audio.pcm.iec958.0 {
+USB-Audio.pcm.iec958.!0 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
@@ -472,7 +472,7 @@ USB-Audio.pcm.iec958.0 {
 	}
 }
 
-USB-Audio.pcm.iec958.1 {
+USB-Audio.pcm.iec958.!1 {
 	@args [ CARD AES0 AES1 AES2 AES3 ]
 	@args.CARD { type string }
 	@args.AES0 { type integer }
-- 
2.52.0

